JavaTraining/BillsBurgers/src/Burger.java

```diff
@@ -1,111 +1,54 @@
 public class Burger {
 
-    protected String name;
-    protected double burgerPrice;
-    protected String breadType;
-    protected boolean meat;
-    protected boolean lettuce;
-    protected double lettucePrice = 0.60;
-    protected boolean cheese;
-    protected double cheesePrice = 1;
-    protected boolean onions;
-    protected double onionsPrice = 0.60;
-    protected boolean tomato;
-    protected double tomatoPrice = 0.75;
+    protected BurgerComponent[] baseComponents;
+    protected BurgerComponent[] additionalComponents;
 
-    public Burger(String name, String breadType, double burgerPrice) {
-        this.name = name;
-        this.breadType = breadType;
-        this.burgerPrice = burgerPrice;
-        this.meat = true;
-    }
-
-    public Burger(String name, int burgerPrice) {
-        this.name = name;
-        this.burgerPrice = burgerPrice;
-        this.meat = true;
-
-    }
-    public void setLettuce(boolean lettuce) {
-        this.lettuce = lettuce;
-    }
-
-    public void setCheese(boolean cheese) {
-        this.cheese = cheese;
-    }
+    protected double basePrice;
 
-    public void setOnions(boolean onions) {
-        this.onions = onions;
+    public Burger() {
+        this.basePrice = 5;
+        this.baseComponents = new BurgerComponent[]{
+                new BurgerComponent("bread"),
+                new BurgerComponent("meat")};
+        this.additionalComponents = new BurgerComponent[]{};
     }
 
-    public void setTomato(boolean tomato) {
-        this.tomato = tomato;
+    public Burger(BurgerComponent[] additionalComponents) {
 
+        this.additionalComponents = additionalComponents;
+        this.basePrice = 5;
+        this.baseComponents = new BurgerComponent[]{
+                new BurgerComponent("bread"),
+                new BurgerComponent("meat")};
     }
 
-    public String getBreadType() {
-        return breadType;
+    public double getBasePrice() {
+        return basePrice;
     }
 
-    public String getName() {
-        return name;
-    }
-
-    public int totalAdditions() {
-
-        int totalAdditions = 0;
-
-        totalAdditions += tomato ? 1 : 0;
-        totalAdditions += lettuce ? 1 : 0;
-        totalAdditions += cheese ? 1 : 0;
-        totalAdditions += onions ? 1 : 0;
-
-        return totalAdditions;
-    }
-
-    public double totalPrice() {
-
-        double totalPrice = burgerPrice;
+    public double additionalComponentsPrice() {
 
-        totalPrice += tomato ? tomatoPrice : 0;
-        totalPrice += lettuce ? lettucePrice : 0;
-        totalPrice += cheese ? cheesePrice : 0;
-        totalPrice += onions ? onionsPrice : 0;
+        double totalPrice = 0;
 
+        for (int i = 0; i < additionalComponents.length; i++) {
+            totalPrice = totalPrice + additionalComponents[i].price();
+        }
         return totalPrice;
-
     }
 
-    public void receiptBreakDown() {
+    public int additionalComponents() {
```

Comment:
is the purpose of this method to count the total number of additional components or is to validate the count doesn't exceed a valid upper limit? Consider the Single Responsibility Principle.